### ISSUE TITLE
[a11y]: Invalid textual form elements use color alone to differentiate focus state

### ISSUE THREAD

**Issue by dchyun** [carbon-design-system/carbon#22785]
### Package

@carbon/react, @carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.59.0

### React version

v1.112.0

### Automated testing tool and ruleset

Manual review

### Description

Various textual form elements, when invalid, are failing [WCAG 1.4.1](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) because they use color alone to differentiate between the default state and focus state, and the difference in color does not meet a color contrast of 3:1.

#### Component affected

- Dropdown
- MultiSelect
- NumberInput
- PasswordInput
- Select
- Slider
- TextArea
- TextInput
- TimePicker

**Default state**

<img width="682" height="232" alt="Image" src="https://github.com/user-attachments/assets/b63a27e7-16ab-43c9-9055-f05a0aadfbe8" />

**Focus state**

<img width="674" height="207" alt="Image" src="https://github.com/user-attachments/assets/7defefba-943d-49af-940d-0a6350be02ed" />

**Color contrast**

<img width="725" height="443" alt="Image" src="https://github.com/user-attachments/assets/e379b12b-3be6-46d1-ac1d-78430494ab63" />

### WCAG 2.1 Violation

Success Criterion 1.4.1 Use of Color

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-textinput--default&args=invalid:!true

### Steps to reproduce

1. View one of the affected component and set it to `invalid`
2. Observe the default state
3. Focus on the element using the keyboard
4. Observe the focus state and how the only changed stylistic element is the color of the border

### Suggested Severity

Medium (Severity 3) = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by chrisgrimley** [comment]
Hi @annawen1, do you need support from the accessibility team on this ticket, or can I remove the 'needs accessibility info label'?

**Comment by annawen1** [comment]
Hi @chrisgrimley, I wanted to check in with you if you had any thoughts on this. If not, I'll tag this with design to see what we can do to differentiate between states.

**Comment by chrisgrimley** [comment]
Hi @annawen1. I don't believe there's any violation here. WCAG 1.4.1 would only apply if you're trying to convey a change in information such as an error. For instance, in the error state example you provided we have the red border AND the error icon. The focus state is compliant as long as we have the border (or another visual cue such as an underline) and it has sufficient contrast. I believe we have both.

Regarding the colours, the contrast ratio calculation is between the border and the adjacent colour (in this case, the white background). The red and the blue against the white have a ratio of 5:1, so this is well within the acceptable range. 

<img width="693" height="434" alt="Image" src="https://github.com/user-attachments/assets/9ac66527-c345-4b46-adba-7f515d9f960d" />
<img width="667" height="419" alt="Image" src="https://github.com/user-attachments/assets/4dffe923-d8c8-446d-b504-67d5b140c29c" />

Let me know if I've overlooked something and we can discuss further if needed.

**Comment by annawen1** [comment]
Thanks @chrisgrimley! I'll close this issue. There is another issue open around differentiation between focus and error states that captures the intent of this issue: https://github.com/carbon-design-system/carbon/issues/11650
